### PR TITLE
parse markdown headings

### PR DESCRIPTION
**Pull request by tpeacock19**
This PR is to include functionality to fold Markdown headings.

### CODE REVIEW

File: treesit-fold.el [hunk 1]
```diff
@@ -1152,6 +1152,45 @@ more information."
...
 
+(defun treesit-fold-markdown-next-heading (node siblings)
+  (or
+   (seq-find
+    (lambda (n)
+      (when-let ((child (treesit-node-child n 0 t)))
+        (and (> (treesit-node-start child) (treesit-node-start node))
+             (treesit-fold--compare-type child "atx_heading"))))
+    (remove node siblings))
+   (treesit-node-get (treesit-node-parent node) '((sibling 1 t)))))
```
Comment (jcs090218) [inline treesit-fold.el:1163]: Can you document this function? Thanks! :)

**Comment by jcs090218** [comment]
Can you paste the screenshot so I know how it looks? 🤔 Thank you!

**Comment by tpeacock19** [comment]
image

**Comment by jcs090218** [comment]
There is an error in this implementation: 🤔 

```elisp
treesit-fold.el:1165:5: Warning: the function ‘treesit-node-get’ is not known to be defined.
```

**Comment by tpeacock19** [comment]
its a helper function defined in `treesit.el`, I'm not sure why it gave you that error, but I've replaced it with a function defined in `treesit.c` that you already have declared

**Review by jcs090218 — APPROVED** [review]

**Comment by jcs090218** [comment]
Thank you!

**Pull request merged**